src/app/clusters/service-area-server/service-area-server.cpp

```diff
@@ -0,0 +1,1273 @@
+    {
+        return aEncoder.EncodeNull();
+    }
+    else
+    {
+        return aEncoder.EncodeList([this](const auto & encoder) -> CHIP_ERROR 
+        {
+            uint32_t              mapIndex = 0;
+            MapStructureWrapper   suppportedMap;
+
+            while (mDelegate->GetSupportedMapByIndex(mapIndex++, suppportedMap))
+            {
+                ReturnErrorOnFailure(encoder.Encode(suppportedMap));
+            }
+            return CHIP_NO_ERROR; 
+        });
+    }
+}
+
+CHIP_ERROR Instance::ReadSelectedLocations(chip::app::AttributeValueEncoder & aEncoder) 
+{
+    if (mDelegate->GetNumberOfSelectedLocations() == 0)
+    {
+        return aEncoder.EncodeNull();
+    }
+    else
+    {
+        return aEncoder.EncodeList([this](const auto & encoder) -> CHIP_ERROR 
+        {
+            uint32_t    locationIndex = 0;
+            uint32_t    selectedLocation;
+
+            while (mDelegate->GetSelectedLocationByIndex(locationIndex++, selectedLocation))
+            {
+                ReturnErrorOnFailure(encoder.Encode(selectedLocation));
+            }
+            return CHIP_NO_ERROR; 
+        });
+    }
+}
+
+CHIP_ERROR Instance::ReadProgress(chip::app::AttributeValueEncoder & aEncoder) 
+{
+    if (mDelegate->GetNumberOfProgressElements() == 0)
+    {
+        return aEncoder.EncodeNull();
+    }
+    else
+    {
+        return aEncoder.EncodeList([this](const auto & encoder) -> CHIP_ERROR 
+        {
+            uint32_t                       locationIndex = 0;
+            Structs::ProgressStruct::Type  progressElement;
+
+            while (mDelegate->GetProgressElementByIndex(locationIndex++, progressElement))
+            {
+                ReturnErrorOnFailure(encoder.Encode(progressElement));
+            }
+            return CHIP_NO_ERROR; 
+        });
+    }
+}
+
+
+//*************************************************************************
+// command handlers
+
+void Instance::HandleSelectLocationsCmd(HandlerContext & ctx, const Commands::SelectLocations::DecodableType & req)
+{
+    ChipLogDetail(Zcl, "HandleSelectLocationsCmd");
+
+    SelectLocationsStatus locationStatus            = SelectLocationsStatus::kSuccess;
+    char locationStatusText[kMaxSizeStatusText + 1] = {'\0'};
+    bool useLocationStatusText                      = false;
+    Status cmdStatus                                = Status::Success;
+
+    uint32_t listIndex = 0;
+    uint32_t oldSelectedLocation;
+
+    // On receipt of this command the device SHALL respond with a SelectLocationsResponse command.
+    Commands::SelectLocationsResponse::Type response;
+
+    size_t numberOfLocations = 0;
+    bool   matchesCurrentSelectedLocations;
+
+    // get number of Selected Locations in command parameter
+    VerifyOrExit((req.newLocations.IsNull() || (CHIP_NO_ERROR == req.newLocations.Value().ComputeSize(&numberOfLocations))), 
+                                    locationStatus = SelectLocationsStatus::kInvalidSet;
```

Comment:
Petru points out that all these variables still use 'location'.  At the moment this is still the SelectLocations command. So part of the cleanup in the renaming PR, just noting.

Reply:
We'll rename these when the spec changes. I'm keeping things as they are for now.